src/libraries/System.Private.DataContractSerialization/src/System/Runtime/Serialization/DataContractResolver.cs

```diff
@@ -8,7 +8,7 @@ namespace System.Runtime.Serialization
 {
     public abstract class DataContractResolver
     {
-        public abstract bool TryResolveType(Type type, Type declaredType, DataContractResolver knownTypeResolver, out XmlDictionaryString typeName, out XmlDictionaryString typeNamespace);
-        public abstract Type ResolveName(string typeName, string typeNamespace, Type declaredType, DataContractResolver knownTypeResolver);
+        public abstract bool TryResolveType(Type type, Type? declaredType, DataContractResolver knownTypeResolver, out XmlDictionaryString? typeName, out XmlDictionaryString? typeNamespace);
+        public abstract Type? ResolveName(string typeName, string? typeNamespace, Type? declaredType, DataContractResolver knownTypeResolver);
```

Comment:
Should `typeName` be nullable? Internal implementation below can handle it as nullable:
https://github.com/dotnet/runtime/blob/bcfd90de2a96f4fbe84448df5989b7fd8b258c8a/src/libraries/System.Private.DataContractSerialization/src/System/Runtime/Serialization/KnownTypeDataContractResolver.cs#L53-L56

Reply:
Should `knownTypeResolver` be nullable? As per [documentation](https://docs.microsoft.com/en-us/dotnet/api/system.runtime.serialization.datacontractresolver.resolvename?view=net-5.0):
> If you want to use the known type resolution logic inside your implementation, a reference to it is passed in as the knownTypeResolver parameter.

In addition, if you look at the code sample within the documentation, it doesn't make use of the parameter.

Reply:
> Should typeName be nullable? Internal implementation below can handle it as nullable:

My thinking is "no" because:
1. The method is named "ResolveName" - callers should need to provide a name to resolve.
2. We never pass `null` into this method

> Should knownTypeResolver be nullable?

The documentation says "a reference to it is passed in as the knownTypeResolver parameter", which tells me that this parameter will not be null when we call it and we never pass null to it. If overrides don't want to use the parameter, that's fine. But if someone does want to use it, they shouldn't need to check it for null since it won't be.